pyluna-pathology/luna/pathology/cli/infer_tile_labels.py

```diff
@@ -1,95 +1,148 @@
-              help='json file with method parameters for loading a saved model.')
-def cli(app_config, datastore_id, method_param_path):
-    """Infer tile labels with a trained model.
-
-    app_config - application configuration yaml file. See config.yaml.template for details.
-
-    datastore_id - datastore name. usually a slide id.
+from luna.common.utils import validate_params
 
-    method_param_path - json file with method parameters for loading a saved model.
+import torch
+from torch.utils.data import DataLoader
+from luna.pathology.common.ml import BaseTorchTileDataset, BaseTorchTileClassifier
 
-    - input_label_tag: job tag used to generate tile labels
+import pandas as pd
+from tqdm import tqdm
 
-    - job_tag: job tag for the inference
+_params = [('input_data', str), ('output_dir', str), ('repo_name', str), ('transform_name', str), ('model_name', str), ('weight_tag', str), ('num_cores', int), ('batch_size', int)]
 
-    - model_package: package to load your model e.g. luna.pathology.models.ov_tissuenet
-
-    - model: model details e.g. {
-          "checkpoint_path": "/path/to/checkpoint",
-          "n_classes": 4
-      }
-
-    - root_path: path to output directory
+@click.command()
+@click.option('-i', '--input_data', required=False,
+              help='path to input data')
+@click.option('-o', '--output_dir', required=False,
+              help='path to output directory to save results')
+@click.option('-rn', '--repo_name', required=False,
+              help="repository name to pull model and weight from, e.g. msk-mind/luna-ml")
+@click.option('-tn', '--transform_name', required=False,
+              help="torch hub transform name")   
+@click.option('-mn', '--model_name', required=False,
+              help="torch hub model name")    
+@click.option('-wt', '--weight_tag', required=False,
+              help="weight tag filename")  
+@click.option('-nc', '--num_cores', required=False,
+              help="Number of cores to use", default=4)  
+@click.option('-bx', '--batch_size', required=False,
+              help="weight tag filename", default=256)    
+@click.option('-m', '--method_param_path', required=False,
+              help='json file with method parameters for tile generation and filtering')
+def cli(**cli_kwargs):
     """
-    init_logger()
-
-    with open(method_param_path, 'r') as yaml_file:
-        method_data = yaml.safe_load(yaml_file)
-    infer_tile_labels_with_datastore(app_config, datastore_id, method_data)
-
-def infer_tile_labels_with_datastore(app_config: str, datastore_id: str, method_data: dict):
-    """Infer tile labels with a trained model.
+    Run with explicit arguments:
+
+    \b
+        infer_tile_labels
+            -i 1412934/data/TileImages
+            -o 1412934/data/TilePredictions
+            -r msk-mind/luna-ml:main 
+            -t tissue_tile_net_transform 
+            -m tissue_tile_net_model_5_class
+            -w main:tissue_net_2021-01-19_21.05.24-e17.pth
```

Comment:
Update the flags to `-rn` `-wt` etc. Is `\b` part of the docstring?

Reply:
The \b preserves the indenting of the text block when showing the CLI help method